src/wp-includes/class-wp-block-bindings-registry.php

```diff
@@ -137,13 +163,48 @@ public function register( string $source_name, array $source_properties ) {
 			return false;
 		}
 
+		// Validate that the uses_context parameter is an array.
+		if ( isset( $source_properties['uses_context'] ) && ! is_array( $source_properties['uses_context'] ) ) {
+			_doing_it_wrong(
+				__METHOD__,
+				__( 'The "uses_context" parameter must be an array.' ),
+				'6.5.0'
+			);
+			return false;
+		}
+
+		if ( ! empty( array_diff( array_keys( $source_properties ), $this->allowed_source_properties ) ) ) {
+			_doing_it_wrong(
+				__METHOD__,
+				__( 'The $source_properties array contains invalid properties.' ),
+				'6.5.0'
+			);
+			return false;
+		}
+
 		$source = new WP_Block_Bindings_Source(
 			$source_name,
 			$source_properties
 		);
 
 		$this->sources[ $source_name ] = $source;
 
+		// Add `uses_context` defined by block bindings sources.
+		add_filter(
+			'get_block_type_uses_context',
+			function ( $uses_context, $block_type ) use ( $source ) {
+				if ( ! in_array( $block_type->name, $this->supported_blocks, true ) || empty( $source->uses_context ) ) {
+					return $uses_context;
+				}
+				// Use array_values to reset the array keys.
+				$merged_uses_context = array_values( array_unique( array_merge( $uses_context, $source->uses_context ) ) );
+
+				return $merged_uses_context;
```

Comment:
Nit, the local variable doesn't seem to add much value:
```suggestion
				return array_values( array_unique( array_merge( $uses_context, $source->uses_context ) ) );
```